lib/shoryuken/manager.rb

```diff
@@ -58,24 +54,28 @@ def stop(options = {})
     def processor_done(queue)
       logger.debug { "Process done for '#{queue}'" }
 
-      dispatch
+      dispatch_async
     end
 
     private
 
-    def dispatch
-      return if @done.true?
-
-      @dispatcher_pool.post(&method(:dispatch_now))
+    def dispatch_async
+      @dispatcher_executor.post(&method(:dispatch_now))
     end
 
     def dispatch_now
-      return if ready.zero?
-      return unless (queue = @polling_strategy.next_queue)
+      return if @done.true?
+
+      begin
+        return if ready.zero?
+        return unless (queue = @polling_strategy.next_queue)
 
-      logger.debug { "Ready: #{ready}, Busy: #{busy}, Active Queues: #{@polling_strategy.active_queues}" }
+        logger.debug { "Ready: #{ready}, Busy: #{busy}, Active Queues: #{@polling_strategy.active_queues}" }
 
-      batched_queue?(queue) ? dispatch_batch(queue) : dispatch_single_messages(queue)
+        batched_queue?(queue) ? dispatch_batch(queue) : dispatch_single_messages(queue)
+      ensure
+        dispatch_async
```

Comment:
We always use Receive Wait Time of > 0, so it will always block if the network is up anyway. The sleep is really just for worst case situations.